Contents/Items/Accessories/SynergyAccessories/GuideToMasterNinja/BaseBookContent.cs

```diff
@@ -20,78 +20,42 @@ internal class PlayerNinjaBook : ModPlayer {
 			NinjaWeeb = false;
 		}
 		public override void UpdateEquips() {
-			if (Player.head == ArmorIDs.Head.NinjaHood && Player.body == ArmorIDs.Body.NinjaShirt && Player.legs == ArmorIDs.Legs.NinjaPants) {
-				NinjaWeeb = true;
-			}
+			if (Player.head == ArmorIDs.Head.NinjaHood && Player.body == ArmorIDs.Body.NinjaShirt && Player.legs == ArmorIDs.Legs.NinjaPants) NinjaWeeb = true;
 		}
 		public override void PostUpdate() {
-			if (GuidetoMasterNinja && GuidetoMasterNinja2) {
-				if (TimerForUltimate >= 60) {
-					if (Player.ownedProjectileCounts[ModContent.ProjectileType<ThrowingKnifeCustom>()] < 1) {
-						for (int i = 0; i < 8; i++) {
-							Projectile.NewProjectile(Player.GetSource_FromThis(), Player.Center, new Vector2(1 + i, 1 + i), ModContent.ProjectileType<ThrowingKnifeCustom>(), Player.HeldItem.damage + 10, 0, Player.whoAmI);
-						}
-					}
-					if (Player.ownedProjectileCounts[ModContent.ProjectileType<ShurikenCustom>()] < 1) {
-						for (int i = 0; i < 5; i++) {
-							Projectile.NewProjectile(Player.GetSource_FromThis(), Player.Center, new Vector2(1 + i, 1 + i), ModContent.ProjectileType<ShurikenCustom>(), Player.HeldItem.damage + 10, 0, Player.whoAmI);
-						}
-					}
-					else {
-						TimerForUltimate = 0;
-					}
+			if (GuidetoMasterNinja && GuidetoMasterNinja2) if (TimerForUltimate >= 60) {
+					if (Player.ownedProjectileCounts[ModContent.ProjectileType<ThrowingKnifeCustom>()] < 1) for (int i = 0; i < 8; i++) Projectile.NewProjectile(Player.GetSource_FromThis(), Player.Center, new Vector2(1 + i, 1 + i), ModContent.ProjectileType<ThrowingKnifeCustom>(), Player.HeldItem.damage + 10, 0, Player.whoAmI);
+					if (Player.ownedProjectileCounts[ModContent.ProjectileType<ShurikenCustom>()] < 1) for (int i = 0; i < 5; i++) Projectile.NewProjectile(Player.GetSource_FromThis(), Player.Center, new Vector2(1 + i, 1 + i), ModContent.ProjectileType<ShurikenCustom>(), Player.HeldItem.damage + 10, 0, Player.whoAmI);
+					else TimerForUltimate = 0;
```

Comment:
This straight up looks disgusting.
First of all, why one-lining?
Second, why no braces here?
Third of all, logic doesn't seems correct at all: if there no throwing knives, spawn 8 throwing knives; if there any shurikens, then reset timer; otherwise spawn 8 shurikens.
Another thing, you could simplify this code by a lot by using variables like so:
```
TimerForUltimate = 0;

int type = 0;
if (Player.ownedProjectileCounts[ModContent.ProjectileType<ThrowingKnifeCustom>() < 1)
  type = ModContent.ProjectileType<ThrowingKnifeCustom>();
else if (Player.ownedProjectileCounts[ModContent.ProjectileType<ShurikenCustom>() < 1)
  type = ModContent.ProjectileType<ShurkenCustom>();
  
if (type != 0) {
  // spawn projs
}
```

Reply:
yep, that code is from the past, anyway I fix the code logic